pkg/shares/share_splitting_test.go

```diff
@@ -0,0 +1,98 @@
+package shares
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+func TestSplitTxs(t *testing.T) {
+	type testCase struct {
+		name string
+		txs  coretypes.Txs
+		want [][]byte
+	}
+	testCases := []testCase{
+		{
+			name: "empty txs",
+			txs:  coretypes.Txs{},
+			want: [][]byte{},
+		},
+		{
+			name: "one small tx",
+			txs:  coretypes.Txs{coretypes.Tx{0xa}},
+			want: [][]uint8{
+				append([]uint8{
+					0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, // namespace id
+					0x1, // info byte
+					0x0, // reserved byte
+					0x1, // unit length of first transaction
+					0xa, // data of first transaction
+				}, bytes.Repeat([]byte{0}, 244)...), // padding
```

Comment:
[optional]
these will technically break when we increase share size, I think we could subtract from the constants here, but considering we are rarely doing that I'm 100% fine with leaving it.

having a test that breaks on purpose if the share size is changed might even be desirable.

Reply:
Agreed, this test will definitely break when we increase share size because the bytes of padding will be incorrect and the reserved byte will likely be two bytes instead of one.

I'm on-board with having these tests break when we introduce share breaking changes.